### PR TITLE
istio-manager API proposal

### PR DESCRIPTION
**Pull request by strekm**
**Description**

Changes proposed in this pull request:

- istio-manager API proposal

**Related issue(s)**
https://github.com/kyma-project/istio/issues/11

### CODE REVIEW

**Review by dariusztutaj — DISMISSED** [review]
/meow

File: docs/xff-proposal.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            maxSurge: 50%
+            maxUnavailable: "0"
+        resources:
+          limits:
+            cpu: 250m
+            memory: 384Mi
+          requests:
+            cpu: 10m
+            memory: 128Mi
+      meshConfig:
```
Comment (pbochynski) [inline docs/xff-proposal.md:37]: Why mesh config is under istiod?

**Review by pbochynski — CHANGES_REQUESTED** [review]
Do you plan to support all [kubernetes settings](https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#KubernetesResourcesSpec)?
It is not clear.
The API design suggests that it is a 1:1 copy of https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#KubernetesResourcesSpec
Please make a decision: do you want to keep it consistent with IstioOperator spec or not.
If you decide to go with spec easily mapped to IstioOperator spec list all supported fields and make it more similar. If you go other way probably it would be better to have API more concise (remove dataplane, controlplane, deployment keywords that do not provide value)

File: docs/xff-proposal.md (hunk 1 above)
Comment (strekm) [reply docs/xff-proposal.md:37]: afaik there's CM for `istiod` created out of `meshConfig`.

**Comment by strekm** [comment]
> Do you plan to support all [kubernetes settings](https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#KubernetesResourcesSpec)? It is not clear. The API design suggests that it is a 1:1 copy of https://istio.io/latest/docs/reference/config/istio.operator.v1alpha1/#KubernetesResourcesSpec Please make a decision: do you want to keep it consistent with IstioOperator spec or not. If you decide to go with spec easily mapped to IstioOperator spec list all supported fields and make it more similar. If you go other way probably it would be better to have API more concise (remove dataplane, controlplane, deployment keywords that do not provide value)

we do not plan to expose everything in Istio CR, only what we want to support. If it makes sense some properties will be copied 1 to 1 like `numTrustedProxies`, some might be slightly moderated like `forwardClientCertDetails`, where we want to expose only some configuration. Finally we will build our dedicated configuration based on generic Istio configurations like `externalProviders`.

File: docs/xff-proposal.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    ingressGateway:
+      deployment:
+        hpa: 
+          maxReplicas: 5
+          minReplicas: 2
+          metrics:
+          - resource:
+              name: cpu
+              targetAverageUtilization: 80
+            type: Resource
```
Comment (videlov) [inline docs/xff-proposal.md:50]: this should be not be there `type: Resource`, correct ?

**Review by videlov — CHANGES_REQUESTED** [review]

File: docs/xff-proposal.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+Status will be implemented per `kyma-operator` requirements.
+
+## Istio CR example
+
+```
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: Istio
+metadata:
+spec:
+  controlplane:
```
Comment (triffer) [inline docs/xff-proposal.md:20]: I think we need to adapt the fields "controlplane" and "dataplane" to use the correct camelCase form, as in the details above.

**Review by pbochynski — APPROVED** [review]

**Review by videlov — APPROVED** [review]

**Pull request merged**